include/eve/module/core/regular/impl/simd/x86/gather.hpp

```diff
@@ -0,0 +1,171 @@
+      constexpr bool d_i64x8  = match(c, int64x8 , uint64x8);
+
+      auto src = alternative(cx, wide<U, N>{}, as<wide<U, N>> {});
+      auto m   = [&]()
+      {
+        auto const mm = expand_mask(cx, as<wide<U, N>> {});
+
+        // Extract the kmask from the logical
+        if constexpr(current_api >= avx512) return mm.storage().value;
+        // On AVX512, there is no AVX2-like mask, so we convert the kmask back to a wide mask
+        // On AVX2, this is no-op
+        else                                return mm.mask();
+      }();
+
+      // 64 bits integer data
+      if      constexpr(i_32x8 && d_i64x8)  return _mm512_mask_i32gather_epi64(src,m, v     ,p,8);
+      else if constexpr(i_32x4 && d_i64x2)  return _mm_mask_i32gather_epi64   (src,pl_t(p),v,m,8);
+      else if constexpr(i_32x4 && d_i64x4)  return _mm256_mask_i32gather_epi64(src,pl_t(p),v,m,8);
+      else if constexpr(i_64x8 && d_i64x8)  return _mm512_mask_i64gather_epi64(src,m, v     ,p,8);
+      else if constexpr(i_64x4 && d_i64x4)  return _mm256_mask_i64gather_epi64(src,pl_t(p),v,m,8);
+      else if constexpr(i_64x2 && d_i64x2)  return _mm_mask_i64gather_epi64   (src,pl_t(p),v,m,8);
+      // 64 bits float data
+      else if constexpr(i_32x8 && c == float64x8) return _mm512_mask_i32gather_pd(src,m,v,p,8);
+      else if constexpr(i_32x4 && c == float64x4) return _mm256_mask_i32gather_pd(src,p,v,m,8);
+      else if constexpr(i_32x4 && c == float64x2) return _mm_mask_i32gather_pd   (src,p,v,m,8);
+      else if constexpr(i_64x8 && c == float64x8) return _mm512_mask_i64gather_pd(src,m,v,p,8);
+      else if constexpr(i_64x4 && c == float64x4) return _mm256_mask_i64gather_pd(src,p,v,m,8);
+      else if constexpr(i_64x2 && c == float64x2) return _mm_mask_i64gather_pd   (src,p,v,m,8);
+      // 32 bits integer data
+      else if constexpr(i_32x16 && d_i32x16)  return _mm512_mask_i32gather_epi32(src,m,v,p,4);
+      else if constexpr(i_32x8  && d_i32x8 )  return _mm256_mask_i32gather_epi32(src,pi_t(p),v,m,4);
+      else if constexpr(i_32x4  && d_i32x4 )  return _mm_mask_i32gather_epi32   (src,pi_t(p),v,m,4);
+      else if constexpr(i_64x8  && d_i32x8 )  return _mm512_mask_i64gather_epi32(src,m,v,p,4);
+      else if constexpr(i_64x4  && d_i32x4 )  return _mm256_mask_i64gather_epi32(src,pi_t(p),v,m,4);
+      else if constexpr(i_64x2  && d_i32x4 )  return _mm_mask_i64gather_epi32   (src,pi_t(p),v,m,4);
+      // 32 bits float data
+      else if constexpr(i_32x16 && c == float32x16) return _mm512_mask_i32gather_ps(src,m,v,p,4);
+      else if constexpr(i_32x8  && c == float32x8 ) return _mm256_mask_i32gather_ps(src,p,v,m,4);
+      else if constexpr(i_32x4  && c == float32x4 ) return _mm_mask_i32gather_ps   (src,p,v,m,4);
+      else if constexpr(i_64x8  && c == float32x8 ) return _mm512_mask_i64gather_ps(src,m,v,p,4);
+      else if constexpr(i_64x4  && c == float32x4 ) return _mm256_mask_i64gather_ps(src,p,v,m,4);
```

Comment:
Unlike the generic implementation, these won't fault, letting one mask off loads beyond the end of an array.

Reply:
So those are ok or should revamp them ?

Reply:
Yes, these are okay